src/huak/project/py_project.rs

```diff
@@ -0,0 +1,201 @@
+    /// use huak::project::Project;
+    ///
+    /// let cwd = env::current_dir().unwrap();
+    /// let project = Project::from(cwd);
+    /// ```
+    pub fn from(path: PathBuf) -> Result<Project, HuakError> {
+        // TODO: Builder.
+        let config = Config::from(&path)?;
+        let venv = match Venv::from(&path) {
+            Ok(v) => v,
+            Err(HuakError::VenvNotFound) => {
+                Venv::new(path.join(venv::DEFAULT_VENV_NAME))
+            }
+            Err(e) => return Err(e),
+        };
+        let manifest_path = &config.manifest().path;
+
+        // Set the root to the directory the manifest file was found.
+        // TODO: This is probably not the right way to do this.
+        let mut root = path;
+        if let Some(parent) = manifest_path.parent() {
+            root = parent.to_path_buf()
+        }
+
+        // We can't know the project type here, but it probably doesn't matter
+        // much. We'll just use the default.
+        let project_type = ProjectType::default();
+
+        Ok(Project {
+            root,
+            project_type,
+            config,
+            venv: Some(venv),
+        })
+    }
+
+    pub fn create_from_template(&self) -> HuakResult<()> {
+        let name = crate::utils::path::parse_filename(&self.root)?.to_string();
+
+        // Create src subdirectory
+        fs::create_dir_all(self.root.join("src"))?;
+
+        if self.project_type == ProjectType::Library {
+            fs::write(
+                &self.root.join("src").join("__init__.py"),
+                "from .my_math import add_one, times_two\n",
+            )?;
+
+            fs::write(&self.root.join("src").join("my_math.py"),
+                      "# welcome to Huak's sample Python library.\ndef add_one(my_number):\
+                      \n\treturn my_number + 1\n\n\ndef times_two(my_number):\n\treturn my_number * 2\n")?;
+
+            fs::write(&self.root.join("test.py"),
+                      "import unittest\nfrom src import add_one\n\n\n\
+                      class TestSum(unittest.TestCase):\n\tdef test_lib_function(self):\n\t\t\
+                      self.assertEqual(add_one(1), 2)\n\n\nif __name__ == '__main__':\n\tunittest.main()\n")?;
+        } else {
+            fs::create_dir_all(self.root.join("src").join(&name))?;
+            fs::write(&self.root.join("src").join(&name).join("main.py"),
+                      "# This is a sample Python script.\ndef print_hi(name):\
+                      \n\tprint(f'Hello, {name}')\n\n\nif __name__ == '__main__':\n\tprint_hi('World')\n")?;
+        }
+        Ok(())
+    }
+
+    /// Create project toml.
+    // TODO: Config implementations?
+    pub fn create_toml(&self) -> HuakResult<Toml> {
+        let mut toml = Toml::default();
+        let name = crate::utils::path::parse_filename(&self.root)?.to_string();
+
+        if matches!(self.project_type, ProjectType::Application) {
+            let entrypoint = format!("{name}:run");
```

Comment:
I was thinking we use a main function as a default entry for app project types.
```suggestion
            let entrypoint = format!("{name}:main");
```

Reply:
This may need an update to a test too.

Reply:
you're probably correct, I didn't look too closely at create_toml function. Really just moved to someplace that made sense to me.

Reply:
yup! it was something i wanted to nit in the pr that added it, but decided I could just follow up at some point to make the change. this would be a good pr to include it in!